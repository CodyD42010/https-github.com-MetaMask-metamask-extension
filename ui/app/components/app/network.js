const Component = require('react').Component
const PropTypes = require('prop-types')
const h = require('react-hyperscript')
const connect = require('react-redux').connect
const classnames = require('classnames')
const inherits = require('util').inherits
const NetworkDropdownIcon = require('./dropdowns/components/network-dropdown-icon')

Network.contextTypes = {
  t: PropTypes.func,
}

module.exports = connect()(Network)


inherits(Network, Component)

function Network () {
  Component.call(this)
}

Network.prototype.render = function () {
  const props = this.props
  const context = this.context
  const networkNumber = props.network
  let providerName, providerNick, providerUrl
  try {
    providerName = props.provider.type
    providerNick = props.provider.nickname || ''
    providerUrl = props.provider.rpcTarget
  } catch (e) {
    providerName = null
  }
  const providerId = providerNick || providerName || providerUrl || null
  let iconName
  let hoverText

  if (providerName === 'mainnet') {
    hoverText = context.t('mainnet')
    iconName = 'ethereum-network'
  } else if (providerName === 'ropsten') {
    hoverText = context.t('ropsten')
    iconName = 'ropsten-test-network'
  } else if (providerName === 'kovan') {
    hoverText = context.t('kovan')
    iconName = 'kovan-test-network'
  } else if (providerName === 'rinkeby') {
    hoverText = context.t('rinkeby')
    iconName = 'rinkeby-test-network'
  } else if (providerName === 'goerli') {
    hoverText = context.t('goerli')
    iconName = 'goerli-test-network'
  } else {
    hoverText = providerId
    iconName = 'private-network'
  }

  return (
    h('div.network-component.pointer', {
      className: classnames({
        'network-component--disabled': this.props.disabled,
        'ethereum-network': providerName === 'mainnet',
        'ropsten-test-network': providerName === 'ropsten',
        'kovan-test-network': providerName === 'kovan',
        'rinkeby-test-network': providerName === 'rinkeby',
        'goerli-test-network': providerName === 'goerli',
      }),
      title: hoverText,
      onClick: (event) => {
        if (!this.props.disabled) {
          this.props.onClick(event)
        }
      },
    }, [
      (function () {
        switch (iconName) {
          case 'ethereum-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#038789', // $blue-lagoon
                nonSelectBackgroundColor: '#15afb2',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('mainnet')),
              h('.network-indicator__down-arrow'),
            ])
          case 'ropsten-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#e91550', // $crimson
                nonSelectBackgroundColor: '#ec2c50',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('ropsten')),
              h('.network-indicator__down-arrow'),
            ])
          case 'kovan-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#690496', // $purple
                nonSelectBackgroundColor: '#b039f3',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('kovan')),
              h('.network-indicator__down-arrow'),
            ])
          case 'rinkeby-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#ebb33f', // $tulip-tree
                nonSelectBackgroundColor: '#ecb23e',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('rinkeby')),
              h('.network-indicator__down-arrow'),
            ])
          case 'goerli-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#3099f2', // $dodger-blue
                nonSelectBackgroundColor: '#ecb23e',
                loading: networkNumber === 'loading',
              }),
              h('.network-name', context.t('goerli')),
              h('.network-indicator__down-arrow'),
            ])
          default:
            return h('.network-indicator', [
              networkNumber === 'loading'
                ? h('span.pointer.network-loading-spinner', {
                  onClick: (event) => this.props.onClick(event),
                }, [
                  h('img', {
                    title: context.t('attemptingConnect'),
                    src: 'images/loading.svg',
                  }),
                ])
                : h('i.fa.fa-question-circle.fa-lg', {
                  style: {
                    color: 'rgb(125, 128, 130)',
                  },
                }),

              h('.network-name', providerName === 'localhost' ? context.t('localhost') : providerNick || context.t('privateNetwork')),
              h('.network-indicator__down-arrow'),
            ])
        }
      })(),
    ])
  )
}
