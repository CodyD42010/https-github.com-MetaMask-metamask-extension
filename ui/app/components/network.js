const Component = require('react').Component
const h = require('react-hyperscript')
const classnames = require('classnames')
const inherits = require('util').inherits
const NetworkDropdownIcon = require('./dropdowns/components/network-dropdown-icon')

module.exports = Network

inherits(Network, Component)

function Network () {
  Component.call(this)
}

Network.prototype.render = function () {
  const props = this.props
  const networkNumber = props.network
  let providerName
  try {
    providerName = props.provider.type
  } catch (e) {
    providerName = null
  }
  let iconName, hoverText

  if (networkNumber === 'loading') {
    return h('span.pointer', {
      style: {
        display: 'flex',
        alignItems: 'center',
        flexDirection: 'row',
      },
      onClick: (event) => this.props.onClick(event),
    }, [
      h('img', {
        title: 'Attempting to connect to blockchain.',
        style: {
          width: '27px',
        },
        src: 'images/loading.svg',
      }),
      h('i.fa.fa-caret-down'),
    ])
  } else if (providerName === 'mainnet') {
    hoverText = 'Main Ethereum Network'
    iconName = 'ethereum-network'
  } else if (providerName === 'ropsten') {
    hoverText = 'Ropsten Test Network'
    iconName = 'ropsten-test-network'
  } else if (parseInt(networkNumber) === 3) {
    hoverText = 'Ropsten Test Network'
    iconName = 'ropsten-test-network'
  } else if (providerName === 'kovan') {
    hoverText = 'Kovan Test Network'
    iconName = 'kovan-test-network'
  } else if (providerName === 'rinkeby') {
    hoverText = 'Rinkeby Test Network'
    iconName = 'rinkeby-test-network'
  } else {
    hoverText = 'Unknown Private Network'
    iconName = 'unknown-private-network'
  }

  return (
    h('div.network-component.pointer', {
      className: classnames('network-component pointer', {
        'network-component--disabled': this.props.disabled,
        'ethereum-network': providerName === 'mainnet',
        'ropsten-test-network': providerName === 'ropsten' || parseInt(networkNumber) === 3,
        'kovan-test-network': providerName === 'kovan',
        'rinkeby-test-network': providerName === 'rinkeby',
      }),
      title: hoverText,
      onClick: (event) => {
        if (!this.props.disabled) {
          this.props.onClick(event)
        }
      },
    }, [
      (function () {
        switch (iconName) {
          case 'ethereum-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#038789', // $blue-lagoon
                nonSelectBackgroundColor: '#15afb2',
              }),
              h('.network-name', {
                style: {
                  color: '#039396',
                }},
              'Main Network'),
              h('i.fa.fa-caret-down.fa-lg'),
            ])
          case 'ropsten-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#e91550', // $crimson
                nonSelectBackgroundColor: '#ec2c50',
              }),
              h('.network-name', {
                style: {
                  color: '#ff6666',
                }},
              'Ropsten Test Net'),
              h('i.fa.fa-caret-down.fa-lg'),
            ])
          case 'kovan-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#690496', // $purple
                nonSelectBackgroundColor: '#b039f3',
              }),
              h('.network-name', {
                style: {
                  color: '#690496',
                }},
              'Kovan Test Net'),
              h('i.fa.fa-caret-down.fa-lg'),
            ])
          case 'rinkeby-test-network':
            return h('.network-indicator', [
              h(NetworkDropdownIcon, {
                backgroundColor: '#ebb33f', // $tulip-tree
                nonSelectBackgroundColor: '#ecb23e',
              }),
              h('.network-name', {
                style: {
                  color: '#e7a218',
                }},
              'Rinkeby Test Net'),
              h('i.fa.fa-caret-down.fa-lg'),
            ])
          default:
            return h('.network-indicator', [
              h('i.fa.fa-question-circle.fa-lg', {
                style: {
                  margin: '10px',
                  color: 'rgb(125, 128, 130)',
                },
              }),

              h('.network-name', {
                style: {
                  color: '#AEAEAE',
                }},
              'Private Network'),
              h('i.fa.fa-caret-down.fa-lg'),
            ])
        }
      })(),
    ])
  )
}
