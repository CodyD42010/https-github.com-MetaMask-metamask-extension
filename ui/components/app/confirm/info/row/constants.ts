export const TEST_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';
