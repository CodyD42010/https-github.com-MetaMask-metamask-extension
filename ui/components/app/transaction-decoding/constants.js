const TX_INSIGHTS_BASE_URI =
  'https://tx-insights-api.metaswap-dev.codefi.network';

export const FETCH_PROJECT_INFO_URI = `${TX_INSIGHTS_BASE_URI}/fetch-project`;
export const FETCH_SUPPORTED_NETWORKS_URI = `${TX_INSIGHTS_BASE_URI}/networks`;
