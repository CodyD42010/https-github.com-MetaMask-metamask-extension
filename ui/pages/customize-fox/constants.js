export const POLISH_OPTIONS = [
  {
    value: 'generative',
    name: 'Generative',
  },
  {
    value: 'ai',
    name: 'AI',
  },
  {
    value: 'editorSelection',
    name: "Editor's selection",
  },
  {
    value: 'default',
    name: 'Your favourite fox',
  },
];

export const EditorSelectionOptions = [
  {
    value: '1',
    name: 'Rose Hue',
  },
  {
    value: '2',
    name: 'Cocoa Chic',
  },
  {
    value: '3',
    name: 'Chromatic Joy',
  },
  {
    value: '4',
    name: 'Woodland Whisper',
  },
  {
    value: '5',
    name: 'Deep Blush',
  },
];
