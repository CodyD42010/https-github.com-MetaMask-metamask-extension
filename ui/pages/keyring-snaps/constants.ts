export const METAMASK_DEVELOPER = 'metamask';
