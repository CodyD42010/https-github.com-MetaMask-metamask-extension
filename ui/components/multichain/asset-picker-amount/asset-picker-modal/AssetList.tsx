import React, { useMemo } from 'react';
import { useSelector } from 'react-redux';
import classnames from 'classnames';
import isEqual from 'lodash/isEqual';
import {
  getNativeCurrencyImage,
  getSelectedAccountCachedBalance,
  getSelectedInternalAccount,
  getShouldHideZeroBalanceTokens,
} from '../../../../selectors';
import {
  getNativeCurrency,
  getTokens,
} from '../../../../ducks/metamask/metamask';
import { useTokenTracker } from '../../../../hooks/useTokenTracker';
import { useUserPreferencedCurrency } from '../../../../hooks/useUserPreferencedCurrency';
import { PRIMARY, SECONDARY } from '../../../../helpers/constants/common';
import { useCurrencyDisplay } from '../../../../hooks/useCurrencyDisplay';
import { AssetType } from '../../../../../shared/constants/transaction';
import { Box } from '../../../component-library';
import TokenCell from '../../../app/token-cell';
import {
  AlignItems,
  BackgroundColor,
  BorderRadius,
  Display,
  FlexWrap,
} from '../../../../helpers/constants/design-system';
import { TokenListItem } from '../..';
import { Asset, Token } from './types';

type AssetListProps = {
  handleAssetChange: (token: Token) => void;
  asset: Asset;
  searchQuery: string;
};

export default function AssetList({
  handleAssetChange,
  asset,
  searchQuery,
}: AssetListProps) {
  const { address: selectedAddress } = useSelector(getSelectedInternalAccount);
  const selectedToken = asset.details?.address;

  const nativeCurrencyImage = useSelector(getNativeCurrencyImage);
  const nativeCurrency = useSelector(getNativeCurrency);
  const shouldHideZeroBalanceTokens = useSelector(
    getShouldHideZeroBalanceTokens,
  );
  const balanceValue = useSelector(getSelectedAccountCachedBalance);
  const tokens = useSelector(getTokens, isEqual);
  const { tokensWithBalances } = useTokenTracker({
    tokens,
    address: selectedAddress,
    hideZeroBalanceTokens: Boolean(shouldHideZeroBalanceTokens),
  });

  const {
    currency: primaryCurrency,
    numberOfDecimals: primaryNumberOfDecimals,
  } = useUserPreferencedCurrency(PRIMARY, { ethNumberOfDecimals: 4 });

  const {
    currency: secondaryCurrency,
    numberOfDecimals: secondaryNumberOfDecimals,
  } = useUserPreferencedCurrency(SECONDARY, { ethNumberOfDecimals: 4 });

  const [, primaryCurrencyProperties] = useCurrencyDisplay(balanceValue, {
    numberOfDecimals: primaryNumberOfDecimals,
    currency: primaryCurrency,
  });

  const [secondaryCurrencyDisplay, secondaryCurrencyProperties] =
    useCurrencyDisplay(balanceValue, {
      numberOfDecimals: secondaryNumberOfDecimals,
      currency: secondaryCurrency,
    });

  const tokenList = useMemo(() => {
    const filteredTokens: Token[] = tokensWithBalances.filter((token: Token) =>
      token.symbol?.toLowerCase().includes(searchQuery.toLowerCase()),
    );

    // prepend native currency to token list if it matches search query
    if (nativeCurrency?.toLowerCase().includes(searchQuery.toLowerCase())) {
      filteredTokens.unshift({
        address: null,
        symbol: nativeCurrency,
        decimals: 18,
        image: nativeCurrencyImage,
        balance: balanceValue,
        string: primaryCurrencyProperties.value,
        type: AssetType.native,
      });
    }

    return filteredTokens;
  }, [
    tokensWithBalances,
    searchQuery,
    nativeCurrency,
    nativeCurrencyImage,
    balanceValue,
    primaryCurrencyProperties.value,
  ]);

  return (
    <Box className="tokens-main-view-modal">
      {tokenList.map((token) => {
        const isSelected =
          token.address?.toLowerCase() === selectedToken?.toLowerCase();
        return (
          <Box
            padding={0}
            gap={0}
            margin={0}
            key={token.symbol}
            backgroundColor={
              isSelected
                ? BackgroundColor.primaryMuted
                : BackgroundColor.transparent
            }
            className={classnames('multichain-asset-picker-list-item', {
              'multichain-asset-picker-list-item--selected': isSelected,
            })}
            onClick={() => handleAssetChange(token)}
          >
            {isSelected ? (
              <Box
                className="multichain-asset-picker-list-item__selected-indicator"
                borderRadius={BorderRadius.pill}
                backgroundColor={BackgroundColor.primaryDefault}
              />
            ) : null}
            <Box
              key={token.address}
              padding={0}
              display={Display.Block}
              flexWrap={FlexWrap.NoWrap}
              alignItems={AlignItems.center}
              style={{ cursor: 'pointer' }}
            >
              <Box marginInlineStart={2}>
                {token.type === AssetType.native ? (
                  <TokenListItem
                    title={nativeCurrency}
                    primary={
                      primaryCurrencyProperties.value ??
                      secondaryCurrencyProperties.value
                    }
                    tokenSymbol={primaryCurrencyProperties.suffix}
                    secondary={secondaryCurrencyDisplay}
                    tokenImage={token.image}
                  />
                ) : (
                  <TokenCell
                    key={token.address}
                    {...token}
                    onClick={() => handleAssetChange(token)}
                  />
                )}
              </Box>
            </Box>
          </Box>
        );
      })}
    </Box>
  );
}
