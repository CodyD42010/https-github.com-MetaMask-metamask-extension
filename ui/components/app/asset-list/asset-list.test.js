import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithProvider } from '../../../../test/jest';
import configureStore from '../../../store/store';
import mockState from '../../../../test/data/mock-state.json';
import { CHAIN_IDS } from '../../../../shared/constants/network';
import AssetList from './asset-list';

// Specific to just the ETH FIAT conversion
const CONVERSION_RATE = 1597.32;
const ETH_BALANCE = '0x041173b2c0e57d'; // 0.0011 ETH ($1.83)

const USDC_BALANCE = '199.4875'; // @ $1
const LINK_BALANCE = '122.0005'; // @ $6.75
const WBTC_BALANCE = '2.38'; // @ $26,601.51

const USDC_CONTRACT = '0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48';
const LINK_CONTRACT = '0x514910771AF9Ca656af840dff83E8264EcF986CA';
const WBTC_CONTRACT = '0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599';

const render = (
  selectedAddress = mockState.metamask.selectedAddress,
  balance = ETH_BALANCE,
  chainId = CHAIN_IDS.MAINNET,
) => {
  const state = {
    ...mockState,
    metamask: {
      ...mockState.metamask,
      providerConfig: { chainId },
      conversionRate: CONVERSION_RATE,
      cachedBalances: {
        [CHAIN_IDS.MAINNET]: {
          [selectedAddress]: balance,
        },
      },
      contractExchangeRates: {
        [USDC_CONTRACT]: 0.00062566,
        [LINK_CONTRACT]: 0.00423239,
        [WBTC_CONTRACT]: 16.66575,
      },
      selectedAddress,
    },
  };
  const store = configureStore(state);
  return renderWithProvider(
    <AssetList onClickAsset={() => undefined} />,
    store,
  );
};

describe('AssetList', () => {
  it('renders AssetList component and shows Refresh List text', () => {
    render();
    expect(screen.getByText('Refresh list')).toBeInTheDocument();
  });

  describe('token fiat value calculations', () => {
    it('calculates the correct fiat account total', () => {
      process.env.MULTICHAIN = 1;

      const _USDC_CONTRACT = USDC_CONTRACT;
      const _USDC_BALANCE = USDC_BALANCE;
      const _LINK_CONTRACT = LINK_CONTRACT;
      const _LINK_BALANCE = LINK_BALANCE;
      const _WBTC_CONTRACT = WBTC_CONTRACT;
      const _WBTC_BALANCE = WBTC_BALANCE;

      jest.mock('../../../hooks/useTokenTracker', () => {
        return {
          useTokenTracker: () => ({
            loading: false,
            tokensWithBalances: [
              {
                address: _USDC_CONTRACT,
                decimals: 6,
                symbol: 'USDC',
                string: _USDC_BALANCE, // $199.36
              },
              {
                address: _LINK_CONTRACT,
                aggregators: [],
                decimals: 18,
                symbol: 'LINK',
                string: _LINK_BALANCE, // $824.78
              },
              {
                address: _WBTC_CONTRACT,
                aggregators: [],
                decimals: 8,
                symbol: 'WBTC',
                string: _WBTC_BALANCE, // $63,381.02
              },
            ],
            error: null,
          }),
        };
      });

      const { container } = render();
      expect(container).toMatchSnapshot();
      expect(screen.getByText('$63,356.88 USD')).toBeInTheDocument();
    });
  });

  describe('buy and receive buttons', () => {
    it('shows Buy and Receive when the account is empty', () => {
      process.env.MULTICHAIN = 1;
      jest.mock('../../../hooks/useTokenTracker', () => ({
        useTokenTracker: () => ({
          loading: false,
          tokensWithBalances: [],
          error: null,
        }),
      }));

      const { queryByText } = render(
        '0xc42edfcc21ed14dda456aa0756c153f7985d8813',
        '0x0',
      );
      expect(queryByText('Buy')).toBeInTheDocument();
      expect(queryByText('Receive')).toBeInTheDocument();
    });

    it('shows only Receive when chainId is not buyable', () => {
      process.env.MULTICHAIN = 1;
      const { queryByText } = render(
        '0xc42edfcc21ed14dda456aa0756c153f7985d8813',
        '0x0',
        '0x8675309', // Custom chain ID that isn't buyable
      );
      expect(queryByText('Buy')).not.toBeInTheDocument();
      expect(queryByText('Receive')).toBeInTheDocument();
    });
  });
});
