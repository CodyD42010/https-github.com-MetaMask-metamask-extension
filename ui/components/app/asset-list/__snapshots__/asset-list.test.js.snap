// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetList token fiat value calculations calculates the correct fiat account total 1`] = `
<div>
  <div
    class="mm-box token-balance-overview mm-box--padding-4 mm-box--display-flex mm-box--justify-content-space-between mm-box--align-items-center"
  >
    <div
      class="mm-box token-balance-overview__balance"
    >
      <h2
        class="mm-box mm-text token-balance-overview__secondary-balance mm-text--heading-lg mm-box--color-text-default"
      >
        $64,382.85
      </h2>
    </div>
    <button
      class="mm-box mm-text mm-button-base mm-button-base--size-md token-balance-portfolio mm-button-secondary mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-4 mm-box--padding-left-4 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent mm-box--rounded-pill mm-box--border-color-primary-default box--border-style-solid box--border-width-1"
    >
      Portfolio
      <span
        class="mm-box mm-icon mm-icon--size-sm mm-box--margin-inline-start-1 mm-box--display-inline-block mm-box--color-inherit"
        style="mask-image: url('./images/icons/export.svg');"
      />
    </button>
  </div>
  <div
    class="mm-box multichain-token-list-item mm-box--display-flex mm-box--gap-4 mm-box--flex-direction-column"
    data-testid="multichain-token-list-item"
  >
    <a
      class="mm-box multichain-token-list-item__container-cell mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-row"
      data-testid="multichain-token-list-button"
      href="#"
    >
      <div
        class="mm-box mm-badge-wrapper mm-box--margin-right-3 mm-box--display-inline-block"
      >
        <div
          class="mm-box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-token mm-avatar-token--with-halo mm-text--body-sm mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
        >
          <img
            aria-hidden="true"
            class="mm-avatar-token__token-image--blurred"
            src="./images/eth_logo.png"
          />
          <img
            alt="ETH logo"
            class="mm-avatar-token__token-image--size-reduced"
            src="./images/eth_logo.png"
          />
        </div>
        <div
          class="mm-box mm-badge-wrapper__badge-container mm-badge-wrapper__badge-container--circular-top-right"
        >
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-border-muted box--border-style-solid box--border-width-1"
          >
            ?
          </div>
        </div>
      </div>
      <div
        class="mm-box multichain-token-list-item__container-cell--text-container mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
        style="flex-grow: 1; overflow: hidden;"
      >
        <div
          class="mm-box mm-box--display-flex mm-box--gap-1 mm-box--justify-content-space-between"
        >
          <div
            class="mm-box mm-box--width-1/3"
          >
            <p
              class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--ellipsis mm-box--color-text-default"
            >
              Ethereum
            </p>
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--text-align-end mm-box--width-2/3 mm-box--color-text-default"
          >
            $1.83 USD
          </p>
        </div>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative"
          data-testid="multichain-token-list-item-value"
        >
          0.0011
           
          ETH
           
        </p>
      </div>
    </a>
  </div>
  <div>
    <div
      class="mm-box multichain-token-list-item mm-box--display-flex mm-box--gap-4 mm-box--flex-direction-column"
      data-testid="multichain-token-list-item"
    >
      <a
        class="mm-box multichain-token-list-item__container-cell mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-row"
        data-testid="multichain-token-list-button"
        href="#"
      >
        <div
          class="mm-box mm-badge-wrapper mm-box--margin-right-3 mm-box--display-inline-block"
        >
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-token mm-avatar-token--with-halo mm-text--body-sm mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-border-default box--border-style-solid box--border-width-1"
          >
            U
          </div>
          <div
            class="mm-box mm-badge-wrapper__badge-container mm-badge-wrapper__badge-container--circular-top-right"
          >
            <div
              class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-border-muted box--border-style-solid box--border-width-1"
            >
              ?
            </div>
          </div>
        </div>
        <div
          class="mm-box multichain-token-list-item__container-cell--text-container mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
          style="flex-grow: 1; overflow: hidden;"
        >
          <div
            class="mm-box mm-box--display-flex mm-box--gap-1 mm-box--justify-content-space-between"
          >
            <div
              class="mm-box mm-box--width-1/3"
            >
              <p
                class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--ellipsis mm-box--color-text-default"
              >
                USDC
              </p>
            </div>
            <p
              class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--text-align-end mm-box--width-2/3 mm-box--color-text-default"
            >
              $199.36 USD
            </p>
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative"
            data-testid="multichain-token-list-item-value"
          >
            199.4875
             
            USDC
             
          </p>
        </div>
      </a>
    </div>
    <div
      class="mm-box multichain-token-list-item mm-box--display-flex mm-box--gap-4 mm-box--flex-direction-column"
      data-testid="multichain-token-list-item"
    >
      <a
        class="mm-box multichain-token-list-item__container-cell mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-row"
        data-testid="multichain-token-list-button"
        href="#"
      >
        <div
          class="mm-box mm-badge-wrapper mm-box--margin-right-3 mm-box--display-inline-block"
        >
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-token mm-avatar-token--with-halo mm-text--body-sm mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-border-default box--border-style-solid box--border-width-1"
          >
            L
          </div>
          <div
            class="mm-box mm-badge-wrapper__badge-container mm-badge-wrapper__badge-container--circular-top-right"
          >
            <div
              class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-border-muted box--border-style-solid box--border-width-1"
            >
              ?
            </div>
          </div>
        </div>
        <div
          class="mm-box multichain-token-list-item__container-cell--text-container mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
          style="flex-grow: 1; overflow: hidden;"
        >
          <div
            class="mm-box mm-box--display-flex mm-box--gap-1 mm-box--justify-content-space-between"
          >
            <div
              class="mm-box mm-box--width-1/3"
            >
              <p
                class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--ellipsis mm-box--color-text-default"
              >
                LINK
              </p>
            </div>
            <p
              class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--text-align-end mm-box--width-2/3 mm-box--color-text-default"
            >
              $824.78 USD
            </p>
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative"
            data-testid="multichain-token-list-item-value"
          >
            122.0005
             
            LINK
             
          </p>
        </div>
      </a>
    </div>
    <div
      class="mm-box multichain-token-list-item mm-box--display-flex mm-box--gap-4 mm-box--flex-direction-column"
      data-testid="multichain-token-list-item"
    >
      <a
        class="mm-box multichain-token-list-item__container-cell mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-row"
        data-testid="multichain-token-list-button"
        href="#"
      >
        <div
          class="mm-box mm-badge-wrapper mm-box--margin-right-3 mm-box--display-inline-block"
        >
          <div
            class="mm-box mm-text mm-avatar-base mm-avatar-base--size-md mm-avatar-token mm-avatar-token--with-halo mm-text--body-sm mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
          >
            <img
              aria-hidden="true"
              class="mm-avatar-token__token-image--blurred"
              src="https://s3.amazonaws.com/airswap-token-images/WBTC.png"
            />
            <img
              alt="WBTC logo"
              class="mm-avatar-token__token-image--size-reduced"
              src="https://s3.amazonaws.com/airswap-token-images/WBTC.png"
            />
          </div>
          <div
            class="mm-box mm-badge-wrapper__badge-container mm-badge-wrapper__badge-container--circular-top-right"
          >
            <div
              class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-network mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-border-muted box--border-style-solid box--border-width-1"
            >
              ?
            </div>
          </div>
        </div>
        <div
          class="mm-box multichain-token-list-item__container-cell--text-container mm-box--display-flex mm-box--flex-direction-column mm-box--width-full"
          style="flex-grow: 1; overflow: hidden;"
        >
          <div
            class="mm-box mm-box--display-flex mm-box--gap-1 mm-box--justify-content-space-between"
          >
            <div
              class="mm-box mm-box--width-1/3"
            >
              <div>
                <div
                  aria-describedby="tippy-tooltip-2"
                  class=""
                  data-original-title="null"
                  data-tooltipped=""
                  style="display: inline;"
                  tabindex="0"
                >
                  <p
                    class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--ellipsis mm-box--color-text-default"
                  >
                    Wrapped Bitcoin
                  </p>
                </div>
              </div>
            </div>
            <p
              class="mm-box mm-text mm-text--body-md mm-text--font-weight-medium mm-text--text-align-end mm-box--width-2/3 mm-box--color-text-default"
            >
              $63,356.88 USD
            </p>
          </div>
          <p
            class="mm-box mm-text mm-text--body-md mm-box--color-text-alternative"
            data-testid="multichain-token-list-item-value"
          >
            2.38
             
            WBTC
             
          </p>
        </div>
      </a>
    </div>
  </div>
  <div
    class="box box--margin-top-4 box--flex-direction-row"
  >
    <div
      class="mm-box multichain-import-token-link mm-box--margin-4 mm-box--margin-bottom-2"
    >
      <div
        class="mm-box mm-box--display-flex mm-box--align-items-center"
      >
        <button
          class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-link mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
          data-testid="import-token-button"
        >
          <span
            class="mm-box mm-icon mm-icon--size-sm mm-box--margin-inline-end-1 mm-box--display-inline-block mm-box--color-inherit"
            style="mask-image: url('./images/icons/add.svg');"
          />
          Import tokens
        </button>
      </div>
      <div
        class="mm-box mm-box--padding-top-2 mm-box--display-flex mm-box--align-items-center"
      >
        <button
          class="mm-box mm-text mm-button-base mm-button-base--size-md mm-button-link mm-text--body-md-medium mm-box--padding-0 mm-box--padding-right-0 mm-box--padding-left-0 mm-box--display-inline-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-primary-default mm-box--background-color-transparent"
          data-testid="refresh-list-button"
        >
          <span
            class="mm-box mm-icon mm-icon--size-sm mm-box--margin-inline-end-1 mm-box--display-inline-block mm-box--color-inherit"
            style="mask-image: url('./images/icons/refresh.svg');"
          />
          Refresh list
        </button>
      </div>
    </div>
  </div>
</div>
`;
